### PR TITLE
Catch and log when project and container image info can't be created

### LINKED ISSUE #12858: Failure to boot in kubernetes in dev mode
This is a regression compared to 1.9.0.Final.

1) Generate a the default REST app
2) add the jib and the minikube extensions
3) add the following properties to the application.properties
````
quarkus.package.type=mutable-jar
quarkus.live-reload.password=changeit 
quarkus.jib.environment-variables.QUARKUS_LAUNCH_DEVMODE=true
````
4) execute `mvn clean package -DskipTests -Dquarkus.kubernetes.deploy=true`

Pointing the browser to the hello endpoint results in
````
java.lang.IllegalStateException: Could not find matching project info read
	at io.dekorate.project.FileProjectFactory.lambda$createInternal$0(FileProjectFactory.java:60)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.dekorate.project.FileProjectFactory.createInternal(FileProjectFactory.java:60)
	at io.dekorate.project.FileProjectFactory.create(FileProjectFactory.java:41)
	at io.quarkus.kubernetes.deployment.KubernetesCommonHelper.createProject(KubernetesCommonHelper.java:77)
	at io.quarkus.kubernetes.deployment.KubernetesCommonHelper.createProject(KubernetesCommonHelper.java:71)
	at io.quarkus.minikube.deployment.MinikubeProcessor.createDecorators(MinikubeProcessor.java:114)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:936)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
````

### PR DESCRIPTION
**Pull request by iocanel**
Fixes: #12858

### CODE REVIEW

**Comment by gastaldi** [comment]
Build failed with:

```
 File '/home/runner/work/quarkus/quarkus/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java' has not been previously formatted.  Please format file and commit before running validation!
```

File: extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java [hunk 1]
```diff
@@ -3,17 +3,24 @@
...
 import io.quarkus.container.spi.ContainerImageInfoBuildItem;
 import io.quarkus.container.spi.ImageReference;
 import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
 
 public class ContainerImageProcessor {
 
+    private static final String UNKNONW_USER = "?";
```
Comment (gsmet) [inline extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java:17]:
```suggestion
    private static final String UNKNOWN_USER = "?";
```

**Review by gsmet — DISMISSED** [review]
Spotted a typo in a constant name.

**Comment by iocanel** [comment]
> Spotted a typo in a constant name.

Thanks for spotting it

**Comment by aloubyansky** [comment]
@iocanel the CI failed to compile
```
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
/home/runner/work/quarkus/quarkus/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageProcessor.java:[39,13] cannot find symbol
  symbol:   variable UNKNONW_USER
  location: class io.quarkus.container.image.deployment.ContainerImageProcessor
```

**Comment by aloubyansky** [comment]
I was just able to compile it locally and it does fix my issue.

**Review by aloubyansky — APPROVED** [review]

**Comment by aloubyansky** [comment]
@gsmet could review it again please? Thanks.

**Pull request merged**